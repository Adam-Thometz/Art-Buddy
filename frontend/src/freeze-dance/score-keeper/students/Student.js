import React from "react";

import { useDispatch } from "react-redux";
import { addPoint, removePoint } from "../../../actions";

import './Student.css';

import Points from "../points/Points";

const Student = ({ student }) => {
  const dispatch = useDispatch();

  const add = () => {
    dispatch(addPoint(student.name));
  };

  const remove = () => {
    if (student.points > 0) {
      dispatch(removePoint(student.name));
    };
  };

  return (
    <div className="Student">
      <p className="Student-name">{student.name}</p>
      <button className="Student-button" onClick={remove}>-</button>
      <button className="Student-button" onClick={add}>+</button>
      <Points points={student.points} />
    </div>
  );
};

export default Student;