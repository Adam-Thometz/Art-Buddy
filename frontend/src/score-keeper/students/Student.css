.Student-name {
  font-size: 2rem;
  margin: 15px;
}

.Student-point-control {
  font-size: 1.2rem;
  margin: 0 auto;
  height: 100px;
}