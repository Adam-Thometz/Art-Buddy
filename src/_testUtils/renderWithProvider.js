import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;