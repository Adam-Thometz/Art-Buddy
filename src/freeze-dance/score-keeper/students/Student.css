.Student-name {
  font-size: 2rem;
}

.Student-button {
  font-size: 1.2rem;
  margin: 0 20px 10px 20px;
  padding: 5px;
  width: 40px;
}