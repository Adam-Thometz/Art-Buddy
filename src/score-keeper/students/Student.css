.Student {
  border: 3px solid #4D4F5E;
  border-radius: 25px;
  margin: 5px;
  padding: 5px;
  box-shadow: 2px 6px 11px 0px #00000040;
}
.Student-name {
  font-size: 2rem;
  margin: 15px;
}

.Student-point-control {
  font-size: 1.2rem;
  margin: 0 auto;
  height: 70px;
  display: flex;
  justify-content: space-between;
}