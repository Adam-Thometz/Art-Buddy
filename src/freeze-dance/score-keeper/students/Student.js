import React from "react";

import { useDispatch } from "react-redux";
import { addPoint, removePoint } from "../../../actions";

import './Student.css'

import Points from "../points/Points";

const Student = ({ student }) => {
  const dispatch = useDispatch();

  const add = () => {
    dispatch(addPoint(student.name));
  };

  const remove = () => {
    dispatch(removePoint(student.name));
  };

  return (
    <div className="Student">
      <p className="Student-name">{student.name}</p>
      <button className="Student-button" onClick={add}>+</button>
      <button className="Student-button" onClick={remove}>-</button>
      <Points points={student.points} />
    </div>
  );
};

export default Student;