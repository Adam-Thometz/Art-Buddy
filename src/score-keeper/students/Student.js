import React from "react";

import { useDispatch } from "react-redux";
import { addPoint, removePoint, removeStudent } from "../../_redux/actions/actions";

import './Student.css';

import Points from "../points/Points";
import Button from "../../_components/button/Button";

const Student = ({ name, color, points }) => {
  const dispatch = useDispatch();

  const remove = () => {
    dispatch(removeStudent(name));
  }

  const plus = () => {
    dispatch(addPoint(name));
  };

  const minus = () => {
    if (points > 0) {
      dispatch(removePoint(name));
    };
  };

  const pointButtonStyles = {
    fontSize: '1.5rem',
    width: '30%'
  }

  return (
    <div className="Student">
      <p className="Student-name" style={{color}}>{name}</p>
      <div className="Student-point-control">
        <Button small otherStyles={pointButtonStyles} colorId={2} onClick={minus}>-</Button>
        <Button small otherStyles={pointButtonStyles} colorId={0} onClick={plus}>+</Button>
      </div>
      <Points points={points} color={color} />
      <Button small colorId={2} onClick={remove}>REMOVE FROM GAME</Button>
    </div>
  );
};

export default Student;