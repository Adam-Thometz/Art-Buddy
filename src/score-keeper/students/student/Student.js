import { useDispatch } from "react-redux";
import { updatePoints } from "_redux/score-keeper/scoreKeeperActions";

import './Student.css';

import addIcon from '_media/_general/add.png';
import subtractIcon from '_media/_general/minus.png';

const Student = ({ name, points }) => {
  const dispatch = useDispatch();

  const add = () => dispatch(updatePoints({ name, instruction: 'add' }));
  const subtract = () => dispatch(updatePoints({ name, instruction: 'subtract' }));

  return (
    <div className="Student">
      <p className="Student-name">{name}</p>
      <div className="Student-point-control">
        <img src={subtractIcon} alt="minus" role='button' onClick={subtract} />
        <p className="Student-points" data-testid='score'>{points}</p>
        <img src={addIcon} alt="add" role='button' onClick={add} />
      </div>
    </div>
  );
};

export default Student;