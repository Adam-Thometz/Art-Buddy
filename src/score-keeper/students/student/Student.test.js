import React from "react";
import renderWithProvider from '_testUtils/renderWithProvider';
import Student from './Student'

describe('Student component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Student />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithProvider(<Student name='Eddie' points={0} />);
    expect(asFragment()).toMatchSnapshot();
  });
});