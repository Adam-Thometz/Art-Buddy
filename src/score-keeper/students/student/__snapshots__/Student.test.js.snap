// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Student"
  >
    <p
      class="Student-name"
    >
      Eddie
    </p>
    <div
      class="Student-point-control"
    >
      <img
        alt="minus"
        role="button"
        src="file"
      />
      <p
        class="Student-points"
        data-testid="score"
      >
        0
      </p>
      <img
        alt="add"
        role="button"
        src="file"
      />
    </div>
  </div>
</DocumentFragment>
`;
